beacon-chain/db/kv/slasher.go

```diff
@@ -340,17 +366,21 @@ func (s *Store) PruneAttestations(ctx context.Context, currentEpoch types.Epoch,
 		return nil
 	}
 	// + 1 here so we can prune everything less than this, but not equal.
-	endPruneEpoch := currentEpoch - types.Epoch(historyLength)
+	endPruneEpoch := currentEpoch - historyLength
 	epochEnc := encodeTargetEpoch(endPruneEpoch, historyLength)
 	return s.db.Update(func(tx *bolt.Tx) error {
-		attBkt := tx.Bucket(attestationRecordsBucket)
-		c := attBkt.Cursor()
-		for k, _ := c.Seek(epochEnc); k != nil; k, _ = c.Prev() {
+		signingRootsBkt := tx.Bucket(attestationDataRootsBucket)
+		attRecordsBkt := tx.Bucket(attestationRecordsBucket)
+		c := signingRootsBkt.Cursor()
+		for k, v := c.Seek(epochEnc); k != nil; k, v = c.Prev() {
 			if !epochPrefixLessThan(k, epochEnc) {
 				continue
 			}
 			slasherAttestationsPrunedTotal.Inc()
-			if err := attBkt.Delete(k); err != nil {
+			if err := signingRootsBkt.Delete(k); err != nil {
+				return err
+			}
+			if err := attRecordsBkt.Delete(v); err != nil {
```

Comment:
We could be trying to delete the same key multiple times here, does that introduce additional overhead?

Reply:
It's no cost. Delete is a no-op if the key doesn't exist, so we should be OK